src/client/serialize/hdf_serialize.c

```diff
@@ -0,0 +1,1033 @@
+	if (rc != 0) {
+		goto out;
+	}
+	entry = &prop->dpp_entries[3];
+	rc = cont_deserialize_prop_uint(file_id, entry, "DAOS_PROP_CO_CSUM");
+	if (rc != 0) {
+		goto out;
+	}
+	entry = &prop->dpp_entries[4];
+	rc = cont_deserialize_prop_uint(file_id, entry,
+					"DAOS_PROP_CO_CSUM_CHUNK_SIZE");
+	if (rc != 0) {
+		goto out;
+	}
+	entry = &prop->dpp_entries[5];
+	rc = cont_deserialize_prop_uint(file_id, entry,
+					"DAOS_PROP_CO_CSUM_SERVER_VERIFY");
+	if (rc != 0) {
+		goto out;
+	}
+	entry = &prop->dpp_entries[6];
+	rc = cont_deserialize_prop_uint(file_id, entry,
+					"DAOS_PROP_CO_REDUN_FAC");
+	if (rc != 0) {
+		goto out;
+	}
+	entry = &prop->dpp_entries[7];
+	rc = cont_deserialize_prop_uint(file_id, entry,
+					"DAOS_PROP_CO_REDUN_LVL");
+	if (rc != 0) {
+		goto out;
+	}
+	entry = &prop->dpp_entries[8];
+	rc = cont_deserialize_prop_uint(file_id, entry,
+					"DAOS_PROP_CO_SNAPSHOT_MAX");
+	if (rc != 0) {
+		goto out;
+	}
+	entry = &prop->dpp_entries[9];
+	rc = cont_deserialize_prop_uint(file_id, entry,
+					"DAOS_PROP_CO_COMPRESS");
+	if (rc != 0) {
+		goto out;
+	}
+	entry = &prop->dpp_entries[10];
+	rc = cont_deserialize_prop_uint(file_id, entry, "DAOS_PROP_CO_ENCRYPT");
+	if (rc != 0) {
+		goto out;
+	}
+	entry = &prop->dpp_entries[11];
+	rc = cont_deserialize_prop_str(file_id, entry, "DAOS_PROP_CO_OWNER");
+	if (rc != 0) {
+		goto out;
+	}
+	entry = &prop->dpp_entries[12];
+	rc = cont_deserialize_prop_str(file_id, entry,
+				       "DAOS_PROP_CO_OWNER_GROUP");
+	if (rc != 0) {
+		goto out;
+	}
+	entry = &prop->dpp_entries[13];
+	rc = cont_deserialize_prop_uint(file_id, entry, "DAOS_PROP_CO_DEDUP");
+	if (rc != 0) {
+		goto out;
+	}
+	entry = &prop->dpp_entries[14];
+	rc = cont_deserialize_prop_uint(file_id, entry,
+					"DAOS_PROP_CO_DEDUP_THRESHOLD");
+	if (rc != 0) {
+		goto out;
+	}
+	entry = &prop->dpp_entries[15];
+	rc = cont_deserialize_prop_uint(file_id, entry,
+					"DAOS_PROP_CO_ALLOCED_OID");
+	if (rc != 0) {
+		goto out;
+	}
+	entry = &prop->dpp_entries[16];
+	/* read acl as a list of strings in deserialize, then convert back to
+	 * acl for property entry 
+ 	 */
+	rc = cont_deserialize_prop_acl(file_id, entry, "DAOS_PROP_CO_ACL");
+	if (rc != 0) {
+		goto out;
+	}
+	*cont_type = prop->dpp_entries[1].dpe_val;
+out:
+	return rc;
+}
+
+int
+deserialize_daos_cont_prop_metadata(char *filename, daos_prop_t *props,
+			  	    uint64_t *cont_type)
+{
+	int			rc = 0;
+	hid_t			file_id;
+
+	file_id = H5Fopen(filename, H5F_ACC_RDONLY, H5P_DEFAULT);
+	if (file_id < 0) {
+		fprintf(stderr, "failed to open hdf5 file");
+		rc = 1;
+		goto out;
+	}
+	rc = cont_deserialize_all_props(&file_id, props, cont_type);
+	if (rc != 0) {
+		fprintf(stderr, "failed to deserialize cont props: "
+			""DF_RC, DP_RC(rc));
+		goto out;
+	}
+out:
+	H5Fclose(file_id);
+	return rc;
+}
+
+static int
+cont_deserialize_usr_attrs(hid_t *file_id, uint64_t *_num_attrs,
+		       	   char ***_names, void ***_buffers, size_t **_sizes)
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs